src/centralstore/incremental/NodeBlock.cpp

```diff
@@ -0,0 +1,261 @@
+    }
+    NodeBlock::nodesDB->flush();  // Sync the file with in-memory stream
+    this->edgeRef = edgeReferenceAddress;
+    return true;
+}
+
+/**
+ * Return a pointer to maching relation block with the given node if found, Else return NULL
+ * **/
+RelationBlock* NodeBlock::searchRelation(NodeBlock withNode) {
+    RelationBlock* found = NULL;
+    RelationBlock* currentRelation = this->getRelationHead();
+    while (currentRelation) {
+        if (currentRelation->source.address == this->addr) {
+            if (currentRelation->destination.address == withNode.addr) {
+                found = currentRelation;
+                break;
+            } else {
+                currentRelation = currentRelation->nextSource();
+            }
+        } else if (!this->isDirected && (currentRelation->destination.address == this->addr)) {
+            if (currentRelation->source.address == withNode.addr) {
+                found = currentRelation;
+                break;
+            } else {
+                currentRelation = currentRelation->nextDestination();
+            }
+        } else {
+            throw "Exception: Unrelated relation block for " + std::to_string(this->addr) +
+                " found in relation block " + std::to_string(currentRelation->addr);
+        }
+    }
+
+    return found;
+}
+
+std::list<NodeBlock> NodeBlock::getEdges() {
+    std::list<NodeBlock> edges;
+    RelationBlock* currentRelation = this->getRelationHead();
+    while (currentRelation != NULL) {
+        NodeBlock* node = NULL;
+        if (currentRelation->source.address == this->addr) {
+            node = NodeBlock::get(currentRelation->destination.address);
+            currentRelation = currentRelation->nextSource();
+        } else if (currentRelation->destination.address == this->addr) {
+            node = NodeBlock::get(currentRelation->source.address);
+            currentRelation = currentRelation->nextDestination();
+        } else {
+            throw "Error: Unrecognized relation for " + std::to_string(this->addr) + " in relation block " +
+                std::to_string(currentRelation->addr);
+        }
+        if (!node) {
+            throw "Error creating node in the relation";
+        }
+        edges.push_back(*node);
+    }
+    return edges;
+}
+
+std::map<std::string, char*> NodeBlock::getAllProperties() {
+    std::map<std::string, char*> allProperties;
+    PropertyLink* current = this->getPropertyHead();
+    while (current) {
+        allProperties.insert({current->name, current->value});
+        PropertyLink* _temp = current->next();
+        delete current;  // To prevent memory leaks
+        current = _temp;
+    }
+    delete current;
+    return allProperties;
+}
+
+NodeBlock* NodeBlock::get(unsigned int blockAddress) {
+    NodeBlock* nodeBlockPointer = NULL;
+    NodeBlock::nodesDB->seekg(blockAddress);
+    unsigned int edgeRef;
+    unsigned int propRef;
+    char usageBlock;
+    char label[6];
```

Comment:
Is it possible to move this 6 to a constant? Why do we use 6 here?

Reply:
It's coming from the definition of Node block structure, We only allow 6 bytes to be stored as node label, Anything longer than that will be stored as a property
![image](https://user-images.githubusercontent.com/3313885/99704712-c445e580-2abe-11eb-808a-341b11b0c2ae.png)

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/1d404d7d8fc79e4d908dcc39667a33b78beeaa26

Reply:
I have added another comment, please check.